adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleInterstitial.m

```diff
@@ -89,98 +92,69 @@ - (void)requestInterstitialAd {
 #pragma mark - GADMediationInterstitialAd Methods
 
 - (void)presentFromViewController:(UIViewController *)rootViewController {
-  NSError *error = nil;
-  if (![VungleSDK.sharedSDK
-               playAd:rootViewController
-              options:GADMAdapterVunglePlaybackOptionsDictionaryForExtras(_adConfiguration.extras)
-          placementID:self.desiredPlacement
-             adMarkup:[self bidResponse]
-                error:&error]) {
-    // Ad not playable.
-    if (error) {
-      [_delegate didFailToPresentWithError:error];
-    }
-  }
+  [_interstitialAd presentWith:rootViewController];
 }
 
 #pragma mark - Private methods
 
 - (void)loadAd {
-  NSError *error = [GADMAdapterVungleBiddingRouter.sharedInstance loadAdWithDelegate:self];
-  if (error) {
-    _adLoadCompletionHandler(nil, error);
-  }
-}
-
-#pragma mark - GADMAdapterVungleDelegate
-
-- (NSString *)bidResponse {
-  return _adConfiguration.bidResponse;
+  _interstitialAd = [[VungleInterstitial alloc] initWithPlacementId:self.desiredPlacement];
+  _interstitialAd.delegate = self;
+  [_interstitialAd load:_adConfiguration.bidResponse];
 }
 
-- (void)initialized:(BOOL)isSuccess error:(nullable NSError *)error {
-  if (!isSuccess) {
-    _adLoadCompletionHandler(nil, error);
-    return;
-  }
-  [self loadAd];
-}
-
-- (void)adAvailable {
-  if (self.isAdLoaded) {
-    // Already invoked an ad load callback.
-    return;
-  }
-  self.isAdLoaded = YES;
+#pragma mark - VungleInterstitialDelegate
 
+- (void) interstitialAdDidLoad:(VungleInterstitial *)interstitial {
   if (_adLoadCompletionHandler) {
     _delegate = _adLoadCompletionHandler(self, nil);
   }
-
-  if (!_delegate) {
-    // In this case, the request for Vungle has been timed out. Clean up self.
-    [GADMAdapterVungleBiddingRouter.sharedInstance removeDelegate:self];
-  }
 }
 
-- (void)adNotAvailable:(nonnull NSError *)error {
-  if (self.isAdLoaded) {
-    // Already invoked an ad load callback.
-    return;
-  }
+- (void) interstitialAdDidFailToLoad:(VungleInterstitial *)interstitial withError:(NSError *)error {
   _adLoadCompletionHandler(nil, error);
 }
 
-- (void)willShowAd {
+- (void) interstitialAdWillPresent:(VungleInterstitial *)interstitial {
```

Comment:
I saw we don't use `interstitial` parameter in the delegate methods. Do we still need this parameter? Thanks a lot. :)